Entomophthora muscae and comparative analyses of entomophthoralean species genomes
===
Comparative analysis focused on _E. muscae_ Berkeley strain. The primary genome annotation process is archived in [Entomophthora_muscae_UCB github repository](https://github.com/zygolife/Entomophthora_muscae_UCB)

This repository contains archive of code used to process the genomes on the [UCR HPCC](https://hpcc.ucr.edu).
The analysis include 
* BUSCO run on species to compare completeness
* data captured for plotting the Nuclei metrics
* [Hotpep](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1625-9) runs for secretome prediction
* Ortholog analysis with OrthoFinder
* Telomere identification with pattern searches
* Repeat Induced Point mutations (RIP) analyses across species genomes
* Quantifying Transposons and Repetitive Element content
* Protein domain content (Pfam, CAZY with dbCAN, MEROPs) profiles across proteomes


The [Supporting_data](/Supporting_data) folder contains archives of transcriptome assemblies and [data used to build multi-strain phylogeny](/Supporting_data/Figure_6A).

==Citation==

Jason E Stajich, Brian Lovett, Emily Lee, Angie M. Macias, Ann E Hajek, Benjamin L de Bivort, Matt T. Kasson, Henrik H. De Fine Licht, Carolyn Elya. Signatures of transposon-mediated genome inflation, host specialization, and photoentrainment in Entomophthora muscae and allied entomophthoralean fungi. _Submitted_.


Curated by Jason Stajich - jason.stajich[AT]ucr.edu

